Add swagger, update readme

jaredkosanovic opened the pull request.

Commit:
update readme

Commit:
update swagger

Commit:
use result objects, add id endpoints

since I moved things into result objects, we should have a self link for each resource object which requires a separate getById endpoint

floppyears commented on swagger.yaml:
```diff
@@ -905,7 +956,15 @@ definitions:
+    properties:
+      links:
+        $ref: "#/definitions/ResultLinks"
+      data:
```
Can a person have more than one of this? Meaning should this be an array or a single object?

Commit:
add relationships, remove pagniation links

I realized pagination links didn't make much sense because we can paginate with the DM API, meaning we would need to cache data somewhere (bleh).

Commit:
array not needed

because 1-1
